File: frontend/src/models/hearing.ts
/** Hearing module request/response types (mirror `backend/app/schemas/hearing.py`). */

import type { Ear, HearingProfile } from '@/models/profile'

export interface HearingTestRequest {
  ear: Ear
  audiogram: Array<{ frequency: number; threshold_db: number }>
  completed_at?: string
}

export interface WaveformPeaks {
  /** [[min, max], ...] per time bucket, for custom-canvas rendering. */
  peaks: number[][]
}

export interface HearingProcessResponse {
  duration_s: number
  sample_rate: number
  original: WaveformPeaks
  processed: WaveformPeaks
  processed_audio_b64: string
  stages: string[]
  meta: Record<string, unknown>
}

export type HearingTestResponse = HearingProfile

File: backend/app/schemas/hearing.py
"""Hearing module request/response schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field

from app.models.profile import AudiogramPoint, Ear, HearingProfile
from app.schemas.common import WaveformPeaks


class HearingTestRequest(BaseModel):
    """A completed hearing assessment, as produced by the in-browser tone test."""

    ear: Ear = "right"
    audiogram: list[AudiogramPoint] = Field(min_length=1)
    completed_at: str | None = None


class HearingProcessResponse(BaseModel):
    """Result of the speech-adaptation pipeline."""

    duration_s: float
    sample_rate: int
    original: WaveformPeaks
    processed: WaveformPeaks
    processed_audio_b64: str
    stages: list[str]
    meta: dict = Field(default_factory=dict)
